core/types/receipt.go

```diff
@@ -610,16 +610,24 @@ func (rs Receipts) DeriveFields(config *params.ChainConfig, hash common.Hash, nu
 		rs[i].Type = txs[i].Type()
 		rs[i].TxHash = txs[i].Hash()
 
-		// Pre-gingerbred the base fee was stored in state, but we don't try to recover it here, since A) we don't have
-		// access to the objects required to get the state and B) retrieving the base fee is quite code heavy and we
-		// don't want to bring that code across from the celo L1 to op-geth. In the celo L1 we would return a nil base
-		// fee if the state was not available, so that is what we do here.
-		if config.IsGingerbread(new(big.Int).SetUint64(number)) {
-			// The post transition CeloDynamicFeeV2Txs set the baseFee in the receipt
-			if rs[i].BaseFee == nil {
-				rs[i].EffectiveGasPrice = txs[i].inner.effectiveGasPrice(new(big.Int), baseFee)
-			} else {
-				rs[i].EffectiveGasPrice = txs[i].inner.effectiveGasPrice(new(big.Int), rs[i].BaseFee)
+		switch rs[i].Type {
+		case LegacyTxType, AccessListTxType:
+			// These are the non dynamic tx types so we can simply set effective gas price to gas price.
+			rs[i].EffectiveGasPrice = txs[i].inner.effectiveGasPrice(new(big.Int), baseFee)
+		default:
+			// Pre-gingerbred the base fee was stored in state, but we don't try to recover it here, since A) we don't have
+			// access to the objects required to get the state and B) retrieving the base fee is quite code heavy and we
+			// don't want to bring that code across from the celo L1 to op-geth. In the celo L1 we would return a nil base
+			// fee if the state was not available, so that is what we do here.
+			if config.IsGingerbread(new(big.Int).SetUint64(number)) {
```

Comment:
I thought we could add a `|| config.isLondon` here, since London IIRC introduced the effectiveGasPrice in the receipt and Celo's equivalent is Gingerbread. Could also be solved in a different way, but setting the effectiveGasPrice for London sounds like a good idea to me.

Reply:
This sounds sensible, but just to check I'm on the same page here, we'd be adding `|| config.isLondon` to support upstream tests that are running without gingerbread being set?

Reply:
Yes, exactly.

Reply:
Done.